src/app/services/journey-communication.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { BehaviorSubject } from 'rxjs';
+
+import { Transaction, TransactionsCommunicationService } from 'transactions-journey';
+import { MakeTransferCommunicationService, Transfer } from 'transfer-journey';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class JourneyCommunicationService implements MakeTransferCommunicationService, TransactionsCommunicationService {
+  private categoryCodeForTransfer = '#c12020';
+  private latestTransaction$$ = new BehaviorSubject<Transaction | undefined>(undefined);
+
+  public latestTransaction$ = this.latestTransaction$$.asObservable();
+
+  constructor(private router: Router) {}
+
+  public makeTransfer({ toAccount, amount }: Transfer): void {
+    this.latestTransaction$$.next({
+      categoryCode: this.categoryCodeForTransfer,
```

Comment:
I do not agree with you.

First of all, the source journey should not be binding with the destination journey and if there will be mapping in the source journey, then it will require the type from the destination one. They will be coupled and I won't to avoid this.

Secondly, Imagine the situation, when tomorrow there will be destination journey 2, then, in your case, I would need to update the API of the source journey to make possible the communication, but the source journey should not be changed after the new consumer of the data appeared, basically source journey should not know if there any consumers or what is the form of the data they needed.

Also, I do not see any sense to make this method flexible, it won't be reusable by any chance and for every new communication, you will need a new method only here.

Reply:
I got it wrong. Now I understand that you are building a transaction here and it is fine. 
I believe it can be even better if you could do that in a separated method so that you can separate concerns.

Reply:
one more thing about this:
**Also, I do not see any sense to make this method flexible, it won't be reusable by any chance and for every new communication, you will need a new method only here.**
this is a sample app and it is supposed to be an example for everybody in the company. So it has to show the best practices possible. A flexible method is good for reusability and also reduces changes in long term and therefore a good practice

Reply:
I agree about extracting mapping, did this here https://github.com/Backbase/golden-sample-app/pull/8/commits/dcaf02bd3fa633b18089e41d60c110c909c18490